### DISCUSSION TITLE
M5Stackのコンパイル時間短縮について

### DISCUSSION THREAD

**Discussion by sesgak** [1ft-seabass/ma-hackathon-qa-and-knowledge-202210#5] [Q&A] [answered]
[ハンズオン事前準備](https://1ft-seabass.github.io/ma-m5stack-handson-202210/00-preparation/02-arduino-test.html)にそってArduinoでコンパイル検証を実施しました。

> スケッチをコンパイルしていますとなって表示されます。通常のパワーの PC であれば 1 ～ 2 分程度でコンパイルできます。

とありますが、初回コンパイルに10分以上かかりました…２回目以降は早くコンパイルできています。
ただ、IDEを起動するたびに初回は10分以上かかっています。初回はどうしても時間がかかるものなのでしょうか？何か回避策があれば教えていただきたいです。

私のPCスペックは以下のとおりです。
> Intel(R) Core(TM) i5-10210U CPU @ 1.60GHz   2.11 GHz
> 16.0 GB (15.8 GB 使用可能)

**Comment by sesgak** [comment, marked as answer]
https://hangstuck.com/arduino-buildpath-fix/
これを試したら、IDE再起動後でもコンパイルは数分で終わりました！解決かもです。

**Reply by 1ft-seabass** [reply]
お試しありがとうございまーす！

はい、この知見使うと、毎回 Arduino IDE の再起動時に破棄されて、はじめからコンパイルするため時間がかかるものが、以前のものを使ってコンパイルするようになるため早くなりますね。

ハンズオンの限定には向いていると思います。@sesgak さんのような、1ボード設定で、だいたいプログラムを開発していくような場合は、とても有効です。

ただ、まっさらな状態からのコンパイルは挙動が読みやすく良い面もあります。この手法が以前のコードの影響で思わぬ副作用があるかもしれないあたりが読みづらいので、もしどうしようもなく挙動がおかしくなったら戻すことも一応頭の片隅に置いておくと良いと思います～。
